### PR TITLE
Load replies faster

### PR DESCRIPTION
**Pull request by mplorentz**
Part of #294. The goal here was to make the replies view load more quickly. In the end I made two changes:
- The main thing slowing down replies loading is just the fact that we are downloading and processing too many events from relays. The real fix for this is to adopt the Gossip/outbox replication model, but that is out of scope for now. One piece of low hanging fruit is to optimize the Discover REQs a little bit, because it currently requests the firehose of notes from all relays, and then we filter client-side to posts within our two hops network. I added an optimization to abort the processing of events from out-of-network authors after JSON parsing but before we save them into Core Data. 
- I also changed the sort order of replies to use the `receivedTime` instead of `createdAt` date. This means that instead of the replies re-ordering themselves for several seconds after you open the view as we download them from relays the order will stay constant. For events with many replies this makes the view stable earlier and allows you to start reading them earlier.

### CODE REVIEW

File: .xcode-version [hunk 1]
```diff
@@ -1 +1 @@
-14.2
+14.3
```
Comment (martindsq) [inline .xcode-version:1]: Thisand changes on main.yml  should be done in a separate PR

File: Nos.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1669,10 +1673,10 @@
 				"INFOPLIST_KEY_UIStatusBarStyle[sdk=iphonesimulator*]" = UIStatusBarStyleDefault;
 				INFOPLIST_KEY_UISupportedInterfaceOrientations_iPad = "UIInterfaceOrientationPortrait UIInterfaceOrientationPortraitUpsideDown UIInterfaceOrientationLandscapeLeft UIInterfaceOrientationLandscapeRight";
 				INFOPLIST_KEY_UISupportedInterfaceOrientations_iPhone = "UIInterfaceOrientationPortrait UIInterfaceOrientationLandscapeLeft UIInterfaceOrientationLandscapeRight";
-				IPHONEOS_DEPLOYMENT_TARGET = 16.2;
+				IPHONEOS_DEPLOYMENT_TARGET = 16.0;
 				LD_RUNPATH_SEARCH_PATHS = "@executable_path/Frameworks";
 				"LD_RUNPATH_SEARCH_PATHS[sdk=macosx*]" = "@executable_path/../Frameworks";
-				MACOSX_DEPLOYMENT_TARGET = 13.1;
+				MACOSX_DEPLOYMENT_TARGET = 13.3;
```
Comment (martindsq) [inline Nos.xcodeproj/project.pbxproj:1679]: These changes should be done in a different PR

File: Nos/Views/RelayView.swift [hunk 3]
```diff
@@ -32,32 +32,59 @@ struct RelayView: View {
     
     var body: some View {
         List {
-            if let relays {
-                Section {
-                    ForEach(relays) { relay in
-                        Text(relay.address ?? Localized.error.string)
-                            .foregroundColor(.textColor)
+            Section {
```
Comment (martindsq) [inline Nos/Views/RelayView.swift:35]: These changes are also in https://github.com/planetary-social/nos/pull/297

**Review by martindsq — APPROVED** [review]

**Comment by mplorentz** [comment]
@martindsq diff is cleaned up now, sorry.
